csrc/device_lower/validation.cpp

```diff
@@ -1326,4 +1326,79 @@ void validate1dTmaLoad(Fusion* fusion) {
   }
 }
 
+void validateScatter(Fusion* fusion) {
+  for (auto sop : ir_utils::getOpsOfType<ScatterOp>(fusion)) {
+    auto in_tv = sop->in()->as<TensorView>();
+    auto out_tv = sop->out()->as<TensorView>();
+
+    // TensorIndexer currently only supports scatter with 1D tensors
+    // due to the non-exactness of non-indexed IDs.
+    NVF_ERROR_EQ(
+        out_tv->getLogicalDomain().size(),
+        1,
+        "Scatter with multi-dimensional tensors is not yet supported: ",
+        sop->toString());
+
+    // Scatter is implemented as an in-place op. To lower it safely, it
+    // needs to be able to alias each other. Here are the conditions to
+    // make sure they are valid input and output tensors.
+
+    NVF_ERROR_EQ(
+        in_tv->uses().size(),
+        1,
+        "Scatter input can only be used by the scatter op: ",
+        toDelimitedString(in_tv->uses()));
+
+    NVF_ERROR_EQ(in_tv->getMemoryType(), out_tv->getMemoryType());
+    NVF_ERROR_EQ(in_tv->getDeviceMesh(), out_tv->getDeviceMesh());
+
+    // Assumes both input and output have set allocation domains
+    NVF_ERROR(
+        in_tv->getMemoryType() == MemoryType::Global || in_tv->hasAllocation(),
```

Comment:
Naive question: why do we expect an explicit allocation domain when TV memory type is not global?

Reply:
Updated the comment